pkg/cmd/roachtest/jobs.go

```diff
@@ -128,17 +128,12 @@ func jobSurvivesNodeShutdown(
 		return nil
 	})
 
-	// Let the current tasks finish and use a new monitor for the restart.
-	// This lets us separate the failure modes.
 	m.Wait()
```

Comment:
Based on my reading, I think this will still fail here since `Wait` will see a dead node event. Previously we were waiting directly on the group directly (`m.g`), restarting the node, then waiting on the monitor. Looking at the monitor's API again, I think what we want is:

```suggestion
	// Expect this wait to run into a dead-node since target was stopped.
	m.ExpectDeath()
	m.Wait()
```

(And keep restarting the node afterwards since the roachtest harness checks that all nodes are available.)

Reply:
Thanks Paul! You are completely right, I made the edit. I should also run this but can't do so until tomorrow.